debian/bullseye/foreman/foreman-plugin.mk.in

```diff
@@ -1,16 +1,40 @@
 BUNDLE = @BUNDLE@
 
 build:
-	cp cache/* /usr/share/foreman/vendor/cache/
-	cp $(PLUGIN).rb /usr/share/foreman/bundler.d/
-	cd /usr/share/foreman && ( \
+	# prepare a local foreman installation based on packages
+	mkdir -p ./usr/share
+	cp -r /usr/share/foreman ./usr/share || true
+
+	# clean up possibly existing state we copied
+	mkdir -p ./var/lib/foreman
+	cp -r /var/lib/foreman/db ./var/lib/foreman || true
+	unlink ./usr/share/foreman/db
+	mkdir ./usr/share/foreman/db
+	cp -rf /usr/share/foreman/db/* ./usr/share/foreman/db/
+	unlink ./usr/share/foreman/tmp
+	mkdir ./usr/share/foreman/tmp
+	rm ./usr/share/foreman/config/initializers/encryption_key.rb
+	rm ./usr/share/foreman/config/database.yml
+
+	# add plugin code to the local foreman installation
+	cp cache/* ./usr/share/foreman/vendor/cache/
+	cp $(PLUGIN).rb ./usr/share/foreman/bundler.d/
+
+	# perform the actual build of the plugin
+	cd ./usr/share/foreman && ( \
 		/usr/bin/foreman-ruby $(BUNDLE) install --local && \
 		if [ "$(PLUGIN_HAS_PACKAGE_JSON)" = "true" ]; then /usr/bin/npm install --no-audit --no-optional --unsafe-perm; fi && \
+		/usr/bin/foreman-ruby $(BUNDLE) exec rake security:generate_encryption_key && \
```

Comment:
What's the reasoning behind removing the old key and then generating a new one?

Reply:
Ah, most probably:

```console
# ls -alh /usr/share/foreman/config/initializers/encryption_key.rb 
lrwxrwxrwx. 1 root root 30 Jun 22 08:02 /usr/share/foreman/config/initializers/encryption_key.rb -> /etc/foreman/encryption_key.rb
```

(A user won't be able to read that)